packages/admin/cms-admin/src/generator/future/generateForm.ts

```diff
@@ -0,0 +1,281 @@
+import { IntrospectionQuery } from "graphql";
+
+import { generateFormField } from "./generateFormField";
+import { FormConfig, GeneratorReturn } from "./generator";
+import { camelCaseToHumanReadable } from "./utils/camelCaseToHumanReadable";
+import { findRootBlocks } from "./utils/findRootBlocks";
+import { generateImportsCode, Imports } from "./utils/generateImportsCode";
+
+export function generateForm(
+    {
+        exportName,
+        baseOutputFilename,
+        targetDirectory,
+        gqlIntrospection,
+    }: { exportName: string; baseOutputFilename: string; targetDirectory: string; gqlIntrospection: IntrospectionQuery },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    config: FormConfig<any>,
+): GeneratorReturn {
+    const gqlType = config.gqlType;
+    const title = config.title ?? camelCaseToHumanReadable(gqlType);
+    const instanceGqlType = gqlType[0].toLowerCase() + gqlType.substring(1);
+    const gqlQueries: Record<string, string> = {};
```

Comment:
This should be gqlDocuments, since it also includes mutations and fragments

Reply:
done